### PR TITLE
feature: allow eduapplauncher to pass `resources` option

### PR DESCRIPTION
**Pull request by fregataa**
Since `createSessionFromTemplate()` method have `resources` parameter and it can overrides resource spec when create session from template, let's just get resource spec from query parameter and pass resource spec through the `resources` parameter.

### CODE REVIEW

File: src/components/backend-ai-change-forgot-password-view.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default class BackendAIChangeForgotPasswordView extends BackendAIPage {
   @property({type: Object}) notification = Object();
   @property({type: Object}) passwordChangeDialog = Object();
   @property({type: Object}) failDialog = Object();
-  @property({type: String}) token = '';
+  @property({type: String}) resources = Object();
```
Comment (agatha197) [inline src/components/backend-ai-change-forgot-password-view.ts]: This is not used. It's okay to remove this line.

File: src/components/backend-ai-edu-applauncher.ts [hunk 2]
```diff
@@ -68,6 +69,12 @@ export default class BackendAiEduApplauncher extends BackendAIPage {
   }
 
   async launch(apiEndpoint: string) {
+    const queryParams = new URLSearchParams(window.location.search);
+    this.resources = {
+      cpu: queryParams.get('cpu'),
+      mem: queryParams.get('mem'),
+      'cuda.shares': queryParams.get('cuda-shares')
```
Comment (agatha197) [inline src/components/backend-ai-edu-applauncher.ts]:
```suggestion
      'cpu': queryParams.get('cpu'),
      'mem': queryParams.get('mem'),
      'cuda.shares': queryParams.get('cuda.shares'),
      'cuda.device': queryParams.get('cuda.device'),
```

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]

File: src/components/backend-ai-change-forgot-password-view.ts (hunk 1 above)
Comment (fregataa) [reply src/components/backend-ai-change-forgot-password-view.ts]: oops...

File: src/components/backend-ai-edu-applauncher.ts (hunk 2 above)
Comment (fregataa) [reply src/components/backend-ai-edu-applauncher.ts]: thank you! I need to learn JS and TS..

**Comment by Sujin-Kim1** [comment]
@fregataa Please resolve the conflicts

**Comment by fregataa** [comment]
> @fregataa Please resolve the conflicts

Thanks, I resolved the conflict!

File: src/components/backend-ai-edu-applauncher.ts [hunk 3]
```diff
@@ -331,7 +339,8 @@ export default class BackendAiEduApplauncher extends BackendAIPage {
       const templateId = sessionTemplates[0].id; // NOTE: use the first template. will it be okay?
       try {
         const mounts = await globalThis.backendaiclient.eduApp.get_mount_folders();
-        const resources = mounts ? {mounts} : {};
+        const projects = await globalThis.backendaiclient.eduApp.get_user_projects();
+        const resources = mounts ? {mounts, ...this.resources, 'group_name': projects[0]['name']} : {...this.resources, 'group_name': projects[0]['name']};
```
Comment (agatha197) [inline src/components/backend-ai-edu-applauncher.ts:348]: this line can occur `TypeError` if the length of the projects is 0.

Comment (fregataa) [reply src/components/backend-ai-edu-applauncher.ts:348]: Could you recommend me how to log or raise exception when `project` is empty?
Cause the backend ensures that every user has a project. it should be an error if it is empty.

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM

**Pull request merged**